### ISSUE TITLE
How to reset the animation to first frame?

### ISSUE THREAD

**Issue by madeuz** [pixijs/pixi-spine#166]
I've tryed "skeleton.setToSetupPose();" and similar but nothing works
Resetting an animation is very important, why there's no method to do it? 
Besides that I think there should be a method to set the animation at desired frame, based on zero to one interpolation (ex. if I want to set the animation in the middle I just write animation.update(.5))
For now update method is very confusing.

**Comment by simzard** [comment]
This worked for me:
// skeleton 
animation.skeleton.setToSetupPose();
animation.state.tracks = [];

**Comment by ivanpopelyshev** [comment]
so, that clears all animations?

**Comment by simzard** [comment]
Yes all the animations are cleared and you have to add them again.
Here is some of my code below to illustrate (I made a SpineSprite class)

I have start, stop, and rewind methods.

In the start method I initialize the animation again. I remove the onCompleteListener after a rewind - since it otherwise keeps on playing, when I play an empty animation - to simulate that it has stopped

SpineSprite.prototype.rewind = function() {
    if (this.onCompleteStateAdded) {
        this.onCompleteStateAdded = false;
        this.sprite.state.listeners = [];
    }
    this.sprite.skeleton.setToSetupPose();
    this.sprite.state.tracks = [];
}

SpineSprite.prototype.start = function(keepRepeating, onCompleteCallback) {
    this.sprite.state.setAnimation(0, this.animation, keepRepeating);
    if (onCompleteCallback) {
        var self = this;
        this.sprite.state.addListener({
            complete: function(entry) {
                onCompleteCallback(entry);
                if (!keepRepeating) {
                    self.stop();
                }
            }
        });
        this.onCompleteStateAdded = true;
    }
}

SpineSprite.prototype.stop = function() {
    this.rewind();
    this.sprite.state.setEmptyAnimation(0);
}

**Comment by simzard** [comment]
```
SpineSprite.prototype.rewind = function() {
    if (this.onCompleteStateAdded) {
        this.onCompleteStateAdded = false;
        //console.log(this.sprite.state);
        this.sprite.state.listeners = [];
    }
    this.sprite.skeleton.setToSetupPose();
    this.sprite.state.tracks = [];
}

SpineSprite.prototype.start = function(keepRepeating, onCompleteCallback) {
    this.sprite.state.setAnimation(0, this.animation, keepRepeating);
    if (onCompleteCallback) {
        var self = this;
        this.sprite.state.addListener({
            complete: function(entry) {
                onCompleteCallback(entry);
                if (!keepRepeating) {
                    self.stop();
                }
            }
        });
        this.onCompleteStateAdded = true;
    }
}

SpineSprite.prototype.stop = function() {
    this.rewind();
    this.sprite.state.setEmptyAnimation(0);
}
```

**Comment by paolotozzo** [comment]
`state.clearTrack(trackIndex);
skeleton.setToSetupPose()`

worked for me

**Comment by ivanpopelyshev** [comment]
Nice, that'll go into readme

**Comment by djmisterjon** [comment]
@simzard 
thank a lot , this avoid me to use 
state.clearTrack(1); state.addAnimation(1, 'wink1', false, ranWait);

this work good for me 
temp1.setEmptyAnimation(0,1);
temp1.setAnimation(0,'idle',true),

**Comment by PlayerArLin** [comment]
I dont know how to reset the animation to first frame,But i know how to reset the animation to last frame.I hope it will help you.
Spine.TrackEntry trackEntry = mSkeletonGraphic.AnimationState.Tracks.Items[index];
trackEntry.Loop = false;
trackEntry.TrackTime = trackEntry.TrackEnd;

**Comment by komali2** [comment]
According to the Spine API that is nearly impossible to find, http://esotericsoftware.com/spine-applying-animations/

The "right" way to do it is to use `addEmptyAnimation` or `setEmptyAnimation`, with a crossfade number, to not have instantaneous transitions. 

```
state.setAnimation(track, "run", true, 0);
state.addEmptyAnimation(track, 1.5, 0);
```

The problem is later, if I try to restart for example the `run` animation on the same track #, it simply won't start. No idea how to fix that.

**Comment by mehlian** [comment]
This solved my problem:
```
//@ts-ignore
this.spine.lastTime = null; // reset timer (private field of spine)
 
this.spine.state.setAnimation(0, animName, loop);
```

**Comment by makc** [comment]
> how to reset the animation to first frame

obj.state.tracks[0].trackTime = 0;
obj.update(0);

but then it just keeps playing so you have to clear the track or something

**Comment by antonioarcuri** [comment]
private void RestartSpineAnimation(SkeletonGraphic skeletonGraphic)
   {
       if (skeletonGraphic != null)
       {
           // Force restart the animation
           skeletonGraphic.AnimationState.ClearTracks();
           skeletonGraphic.AnimationState.SetAnimation(0, skeletonGraphic.startingAnimation, skeletonGraphic.startingLoop);
       }
   }

**Comment by antonioarcuri** [comment]
private void RestartSpineAnimation(SkeletonGraphic skeletonGraphic)
   {
       if (skeletonGraphic != null)
       {
           // Force restart the animation
           skeletonGraphic.AnimationState.ClearTracks();
           skeletonGraphic.AnimationState.SetAnimation(0, skeletonGraphic.startingAnimation, skeletonGraphic.startingLoop);
       }
   }
